src/states/node.rs

```diff
@@ -2640,8 +2683,156 @@ mod tests {
             self.node_state().has_resource_proof_candidate()
         }
 
-        fn is_peer_valid(&self, public_id: &PublicId) -> bool {
-            self.node_state().chain().is_peer_valid(public_id)
+        fn has_candidate_info(&self) -> bool {
+            self.node_state().peer_mgr().has_candidate_info()
+        }
+
+        fn is_candidate_a_valid_peer(&self) -> bool {
+            self.node_state()
+                .chain()
+                .is_peer_valid(self.candidate_info.new_full_id.public_id())
+        }
+
+        fn dispatch_routing_message(
+            &mut self,
+            routing_msg: RoutingMessage,
+        ) -> Result<(), RoutingError> {
+            unwrap!(self.machine.current_mut().node_state_mut())
+                .dispatch_routing_message(routing_msg, &mut self.ev_buffer)
+        }
+
+        fn handle_direct_message(
+            &mut self,
+            msg: (DirectMessage, PublicId),
+        ) -> Result<(), RoutingError> {
+            let _ = unwrap!(self.machine.current_mut().node_state_mut()).handle_direct_message(
+                msg.0,
+                msg.1,
+                &mut self.ev_buffer,
+            )?;
+            Ok(())
+        }
+
+        fn expect_candidate_payload(&self) -> ExpectCandidatePayload {
+            ExpectCandidatePayload {
+                old_public_id: *self.candidate_info.old_full_id.public_id(),
+                old_client_auth: Authority::Client {
+                    client_id: *self.candidate_info.old_full_id.public_id(),
+                    proxy_node_name: *self.candidate_info.old_proxy_id.public_id().name(),
+                },
+                message_id: self.candidate_info.message_id,
+                dst_name: XorName([0; XOR_NAME_LEN]),
+            }
+        }
+
+        fn online_payload(&self) -> (PublicId, OnlinePayload) {
+            let client_auth = Authority::Client {
+                client_id: *self.candidate_info.new_full_id.public_id(),
+                proxy_node_name: *self.candidate_info.new_proxy_id.public_id().name(),
+            };
+            (
+                *self.candidate_info.new_full_id.public_id(),
+                OnlinePayload {
+                    client_auth,
+                    old_public_id: *self.candidate_info.old_full_id.public_id(),
+                },
+            )
+        }
+
+        fn offline_payload(&self) -> PublicId {
+            *self.candidate_info.new_full_id.public_id()
+        }
+
+        fn purge_payload(&self) -> PublicId {
+            *self.candidate_info.old_full_id.public_id()
+        }
+
+        fn expect_candidate_rpc(&self) -> RoutingMessage {
```

Comment:
A nitpick, but we don't use the term "RPC" anywhere in routing codebase - instead, we use "message". I'd stick to that term to not unnecessarilly introduce new terminology.